code/site/components/com_pages/template/helper/snippet.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+class ComPagesTemplateHelperSnippet extends ComPagesTemplateHelperAbstract
+{
+    private $__snippets = array();
+
+    public function define(string $name, string $snippet, $overwrite = false)
+    {
+        if(!isset($this->__snippets[$name]) || $overwrite) {
+            $this->__snippets[$name] = $snippet;
+        }
+    }
+
+    public function expand(string $name, array $variables = array())
+    {
+        $result = false;
+
+        if(isset($this->__snippets[$name]))
+        {
+            $snippet = $this->__snippets[$name];
+
+            //Use the stream buffer to evaluate the partial
+            $str = "<?php \n return <<<SNIPPET\n$snippet\nSNIPPET;\n";
+
+            $buffer = $this->getObject('filesystem.stream.factory')->createStream('koowa-buffer://temp', 'w+b');
+            $buffer->truncate(0);
+            $buffer->write($str);
+
+            extract($variables, EXTR_OVERWRITE);
+
+            $result = include $buffer->getPath();
+
+            //Cleanup whitespace
+            $result = str_replace(array(' >', ' "'), array('>', '"'), $result);
```

Comment:
@johanjanssens would this not cleanup the quotes in string literals too? say you have a string like `click "subscribe" to proceed` in your snippet. This would turn it into `click"subscribe" to proceed` right?

Reply:
True, this needs a more advanced regex, haven’t gotten to that yet. Reason for this is because it generated clean html which was a request from Robin.

Reply:
@johanjanssens Once you encounter this problem it's very hard to track it down and work around it so I would leave this out.

Reply:
@ercanozkaya Uncommented it for now. Could you provide a regex that could track this but inside < > only? That way i can see clean it.

Reply:
Optimised this with a simple regex. I’m now only stripping single whitespace before “ and > in html tags. See: https://github.com/joomlatools/joomlatools-pages/pull/147/commits/33219b35af055f394a977998c46fe86ec6118139